src/views/AddItem.jsx

```diff
@@ -1,3 +1,112 @@
-export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+import { useState } from 'react';
+import { addItem } from '../api/firebase';
+
+export function AddItem({ listToken }) {
+	const soon = 7;
+	const kindaSoon = 14;
+	const notSoon = 30;
+
+	const [itemName, setItemName] = useState('');
+	const [nextPurchase, setNextPurchase] = useState(0);
+	const [submissionConfirmation, setsubmissionConfirmation] = useState('');
+
+	const handleChange = (e) => {
+		setItemName(e.target.value);
+	};
+
+	const submitForm = (e) => {
+		e.preventDefault();
+
+		//form validation
+		if (itemName === '') {
+			alert('Please specify the name of the item');
+			return;
+		}
+
+		if (nextPurchase === 0) {
+			alert(
+				"Please select how soon you'll be purchasing the item again (soon/Kinda Soon/Not Soon",
+			);
+			return;
+		}
+
+		let itemData = {
+			itemName: itemName,
+			daysUntilNextPurchase: nextPurchase,
+		};
+
+		addItem(listToken, itemData)
+			.then(() => {
+				//clear form
+				setItemName('');
+				setNextPurchase(0);
+
+				// put a note that the form was submitted (and erase after 5 seconds)
+				setsubmissionConfirmation(
+					`${itemName} was added to your shopping list.`,
+				);
+				setTimeout(setsubmissionConfirmation, 5000);
+			})
+			.catch(() => {
+				setsubmissionConfirmation(
+					`There was a problem adding your item, please try again.`,
+				);
```

Comment:
you could wrap this in a try/catch block and pass `error` as a parameter to the catch statement in the case that we get unpredictable errors

Reply:
I think that we're already taking care of the error by using the .catch() function. Because this is an async function, we would need to use try{ await addItem() { catch {} and turn our submitForm function into an async function.
It feels like a lot of changes to catch an error that the user doesn't need to know about (and that we wouldn't be handling anyways)...

Reply:
@spittel @meganesu @xavierbellhaddon This code is meeting the AC as is, but I'd love your input on this if you have some time... How would you handle it?

Reply:
Great question!

You're right that your `.catch()` block will catch any errors that get thrown in your async function.

Although currently, your `.catch()` block isn't displaying the actual contents of the error anywhere, which makes it hard to debug what's going wrong when an error is thrown.

In a production app, I've seen codebases that log these errors using a specific monitoring tool (New Relic, for example), so then devs can get more insights into what kinds of errors users may be running into. In this case though, it's probably good enough to just `console.log` out the error message, like so:

```suggestion
			.then(() => {
				//clear form
				setItemName('');
				setNextPurchase(0);

				// put a note that the form was submitted (and erase after 5 seconds)
				setsubmissionConfirmation(
					`${itemName} was added to your shopping list.`,
				);
				setTimeout(setsubmissionConfirmation, 5000);
			})
			.catch((error) => {
				setsubmissionConfirmation(
					`There was a problem adding your item, please try again.`,
				);
				console.log(error);
```